File: programers/1-3진법뒤집기.py
def solution(n):
    answer = []
    while n:
        n, rest = divmod(n, 3)
        answer.append(rest)
        
    for i in range(len(answer)):
        if answer[i] != 0:
            answer[i] = answer[i]*(3**(len(answer) - 1 - i))
    return sum(answer)

File: programers/1-3진법뒤집기.js
function transform(n, notation) {
  let number = n;
  const transformArr = [];
  while (number / notation > 0) {
    transformArr.push(number % notation);
    number = parseInt(number / notation);
  }
  return transformArr;
}

function solution(n) {
  let answer = 0;
  const notation = 3;
  const transformArr = transform(n, notation);
  const transformArrLen = transformArr.length;
  for (let exponent = 0; exponent < transformArrLen; exponent++) {
    answer +=
      transformArr[exponent] *
      Math.pow(notation, transformArrLen - 1 - exponent);
  }
  return answer;
}
